converters/toZigbee.js

```diff
@@ -4982,6 +4982,38 @@ const converters = {
             }
         },
     },
+    wiser_vact_calibrate_valve: {
+        key: ['calibrate_valve'],
+        convertSet: async (entity, key, value, meta) => {
+            return {state: {'calibrate_valve': value}};
+        },
+    },
+    wiser_sed_zone_mode: {
+        key: ['zone_mode'],
+        convertSet: async (entity, key, value, meta) => {
+            return {state: {'zone_mode': value}};
+        },
+    },
+    wiser_sed_occupied_heating_setpoint: {
+        key: ['occupied_heating_setpoint'],
+        convertSet: async (entity, key, value, meta) => {
+            return {state: {'occupied_heating_setpoint': value}};
+        },
+    },
+    wiser_sed_thermostat_local_temperature_calibration: {
+        key: ['local_temperature_calibration'],
+        convertSet: async (entity, key, value, meta) => {
+            globalStore.putValue(entity, 'localTemperatureCalibrationUpdated', true);
```

Comment:
Instead of saving this in the globalStore and responding to the read request with a read response in fromZigbee there is now a better way to do this in the latest-dev branch.

You can replace this line with `endpoint.saveClusterAttributeKeyValue('hvacThermostat', {localTemperatureCalibration: value})` and zigbee-herdsman will automatically respond upon read.

Reply:
That's a nice feature. :-) It should allow me to refactor alot of these, in principle I can do away with all/most of `wiser_smart_thermostat_client`

However some of them need to be pushed out to the device (via an attribute write) if/when the value is updated from z2m (like the example you highlight above).  Is there a best practice for these type of attribute updates for sleepy end devices?

Reply:
I just checked, and it seems that 1a2097a8773cf521fb727aa40c194b63a8a1c1ac only supports named attributes, is this by design, or could it be extended to support number only attributes?

Reply:
> However some of them need to be pushed out to the device (via an attribute write) if/when the value is updated from z2m (like the example you highlight above). Is there a best practice for these type of attribute updates for sleepy end devices?

Just implemented this, see https://github.com/Koenkk/zigbee-herdsman-converters/pull/2558/files#r635451761

Reply:
> I just checked, and it seems that Koenkk/zigbee-herdsman@1a2097a8773cf521fb727aa40c194b63a8a1c1ac only supports named attributes, is this by design, or could it be extended to support number only attributes?

If needed I can extend it.

Reply:
Hi @Koenkk thanks for the feedback, I have worked a bit on refactoring it (see the following commits), I may be holding one of the APIs wrong or else there may be some underlying issue, the behavior isn't quite as expected.

Reply:
> If needed I can extend it.

I have given the example if we don't below, as it may be an unnecessary complication of the core attribute handling, for my one edge case. The workaround below gets the job done, albeit in a slightly clunky way.

Reply:
I'll keep hunting the root cause, there does seem to be something quite suspect in the log entry though:

`zigbee-herdsman:controller:endpoint Command 0x00124b000426a8a0/11 hvacThermostat.wiserSmartCalibrateValve({"srcEndpoint":11,"disableDefaultResponse":true,"sendWhenActive":true}, {"sendWhenActive":false,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) +6s`

Reply:
> Hi @Koenkk thanks for the feedback, I have worked a bit on refactoring it (see the following commits), I may be holding one of the APIs wrong or else there may be some underlying issue, the behavior isn't quite as expected.

These `sendWhenAcive` issues are now resolved, either by fixes on your side, or by me not being foolish and forgetting parameters.  Just the item below.